src/Simulate.cpp

```diff
@@ -1,13 +1,86 @@
 #include "Simulate.h"
 
-Simulate::Simulate() {
+#include <ctype.h>
+#include <PxPhysicsAPI.h>
+#include <vehicle/PxVehicleUtil.h>
+#include <iostream>
+
+using namespace physx;
+
+PxDefaultAllocator		gAllocator;
+PxDefaultErrorCallback	gErrorCallback;
+
+PxFoundation* gFoundation = NULL;
+PxPhysics* gPhysics = NULL;
+
+PxDefaultCpuDispatcher* gDispatcher = NULL;
+PxScene* gScene = NULL;
+
+PxMaterial* gMaterial = NULL;
 
+PxPvd* gPvd = NULL;
+
+Simulate::Simulate() {
+	initPhysics();
 }
 
 Simulate::~Simulate() {
 
 }
 
-void Simulate::simulateStep(Vehicle vehicles[], Arena arena, Pickup pickups[]) {
+//////////////////////////////////////////////////////////////////////////////////////////////////
+void Simulate::initPhysics()
+{
+	gFoundation = PxCreateFoundation(PX_PHYSICS_VERSION, gAllocator, gErrorCallback);
+
+	gPvd = PxCreatePvd(*gFoundation);
+	PxPvdTransport* transport = PxDefaultPvdSocketTransportCreate("127.0.0.1", 5425, 10);
+	gPvd->connect(*transport, PxPvdInstrumentationFlag::eALL);
+
+	gPhysics = PxCreatePhysics(PX_PHYSICS_VERSION, *gFoundation, PxTolerancesScale(), true, gPvd);
+
+	PxSceneDesc sceneDesc(gPhysics->getTolerancesScale());
+	sceneDesc.gravity = PxVec3(0.0f, -9.81f, 0.0f);
+
+	gDispatcher = PxDefaultCpuDispatcherCreate(2);
+	sceneDesc.cpuDispatcher = gDispatcher;
+	sceneDesc.filterShader = PxDefaultSimulationFilterShader;
+
+	gScene = gPhysics->createScene(sceneDesc);
+	PxPvdSceneClient* pvdClient = gScene->getScenePvdClient();
+	if (pvdClient)
+	{
+		pvdClient->setScenePvdFlag(PxPvdSceneFlag::eTRANSMIT_CONSTRAINTS, true);
+		pvdClient->setScenePvdFlag(PxPvdSceneFlag::eTRANSMIT_CONTACTS, true);
+		pvdClient->setScenePvdFlag(PxPvdSceneFlag::eTRANSMIT_SCENEQUERIES, true);
+	}
+
+	std::cout << "PhysX Initialized" << std::endl;
+}
+
+void Simulate::stepPhysics()
+{
+	gScene->simulate(1.0f / 60.0f);
+	gScene->fetchResults(true);
+}
+
+void Simulate::cleanupPhysics()
+{
+	gScene->release();
+	gDispatcher->release();
+	gPhysics->release();
+	if (gPvd)
+	{
+		PxPvdTransport* transport = gPvd->getTransport();
+		gPvd->release();	gPvd = NULL;
+		transport->release();
+	}
+	gFoundation->release();
+
+	printf("SnippetHelloWorld done.\n");
```

Comment:
Wanna remove this line?

`cleanupPhysics()` isn't called anywhere but I can handle that tomorrow when I do a little bit of reorganizing of the these classes. I will clean up PhysX when cleaning up GLFW.